modin/core/dataframe/pandas/dataframe/dataframe.py

```diff
@@ -1161,12 +1161,22 @@ def _take_2d_positional(
                     extra_log="Mask takes only list-like numeric indexers, "
                     + f"received: {type(indexer)}",
                 )
+                if isinstance(indexer, list):
+                    indexer = np.array(indexer, dtype=np.int64)
             indexers.append(indexer)
         row_positions, col_positions = indexers
 
         if col_positions is None and row_positions is None:
             return self.copy()
 
+        # quite fast check that allows skip sorting
+        must_sort_row_pos = row_positions is not None and not np.all(
+            row_positions[1:] >= row_positions[:-1]
```

Comment:
this check was copied from [`_maybe_reorder_labels`](https://github.com/modin-project/modin/blob/097ea527c8e3f099e1f252b067a1d5eb055ad0b5/modin/core/dataframe/pandas/dataframe/dataframe.py#L1294) and it indeed works pretty fast:
```python
import numpy as np
from timeit import default_timer as timer

arr = np.random.randint(0, 100_000_000, size=100_000_000)

t1 = timer()
must_sort_col_pos = np.all(arr[1:] >= arr[:-1])
print(timer() - t1) # 0.09s
```